src/mono/wasm/build/WasmApp.targets

```diff
@@ -132,7 +151,11 @@
     </ItemGroup>
   </Target>
 
-  <Target Name="_WasmBuildApp" BeforeTargets="WasmBuildApp" DependsOnTargets="$(WasmBuildAppDependsOn)">
+  <Target Name="_WasmCoreBuild" BeforeTargets="WasmBuildApp" DependsOnTargets="$(WasmBuildAppDependsOn)" />
```

Comment:
From what I understand, 
`_WasmCoreBuild` sets this order: `$(WasmBuildAppDependsOn)` -> `_WasmCoreBuild` -> `WasmBuildApp`
`WasmBuildApp` sets this order: `$(WasmBuildAppAfterThisTarget)` -> `WasmBuildApp`
Why would we need `_WasmCoreBuild`?

Can the two targets be condensed into `<Target Name="WasmBuildApp" AfterTargets="$(WasmBuildAppAfterThisTarget)" DependsOnTargets="$(WasmBuildAppDependsOn)" />`

Reply:
So, we have:

1. `WasmBuildApp`, which runs *after* Publish
     - this is the target that the user can add targets to run before/after, like for a `PrepareForWasmBuild` target

2. Before this PR, we had `_WasmBuildApp`, which depends on bunch of targets that do AOT, generate run scripts etc.
    - And then this task itself does the bundle generation

3. But in this PR, we want to make the bundle generation step optional, and still run the other targets. But adding a condition for that would skip the DependsOn stuff too.
   - so, we add a `_WasmCoreBuild` which very explicitly lists all the targets, and the order in which to run them
   - that way, we can skip individual targets as required

I can add some of this to the comment.